### ISSUE TITLE
Armv8-M

### ISSUE THREAD

**Issue by tomasjakubik** [adamgreen/CrashCatcher#18]
Now the M33s are more and more common. Any plans for Armv8-M?
We will try with the Armv7-M config, but would you have some advice that might be useful? I hope that if we won't use the secure modes, it could be the same.

**Comment by adamgreen** [comment]
I don't currently have any plans to implement ARMv8-M support but I would be happy to review and merge a PR for such support.

It would be cool to start with just the non-secure stuff and see if it just works as is. A person could then invest the time in reading the ARMv8-M documentation to see exactly what changes they would need to make to have it work in the secure mode as well. If it ends up adding registers to cover the additional secure mode then CrashDebug would probably need to be modified as well to expose the additional registers on such ARMv8-M dumps and not expose them on all other types of dumps (ARMv6/7-M, etc).

**Comment by Olstyle** [comment]
Looks like Percepio did the work but didn't share it here:
https://percepio.com/from-crashes-to-clarity-whats-new-in-percepio-detect-2025-2/
as far as i understand your Apache License allows that, right?

**Comment by johankraft** [comment]
Thanks for noticing. Yes, our Detect product includes an enhanced version of CrashCatcher, for example Armv8-M support, IAR support and we have also added a function that lets users trigger a dump from their code (DFM_TRAP) by simulating a fault exception, although this is part of the separate DFM library. Our version is found in https://github.com/percepio/demos (see PercepioLibs folder) under the same Apache 2.0 license. Our code is however developed with Percepio Detect in mind and it is not intended or tested for stand-alone use.

**Comment by adamgreen** [comment]
Thanks @Olstyle and @johankraft for bringing this to my attention.

I plan to do some work with an ARMv8-M based microcontroller (RP2350) next year for another project so hopefully I can take a look at incorporating any required changes into CrashDump at that time. I will definitely take a peek at the **Percepio** link again at that time. It looks like the main change is to disable the MSPLIM checks before switching to CrashDump's own stack space.

**Comment by johankraft** [comment]
Yes, I beleive that was the main change (both MSPLIM and PSPLIM). Otherwise you might trip the stack overflow supervision when updating SP to the CC stack, resulting in usagefault exceptions. Learned that the hard way.
